src/grammar.rs

```diff
@@ -1118,12 +1064,12 @@ impl Grammar {
     pub fn parse_formula(
         &self,
         symbol_iter: &mut dyn Iterator<Item = Symbol>,
-        expected_typecodes: Box<[&TypeCode]>,
+        expected_typecodes: Box<[TypeCode]>,
```

Comment:
This is a change in the API of the library, but for the better.

Reply:
Speaking of which, I think the version of mm-knife is way off for semver purposes. We should definitely be on 0.x.y, we don't have any current API users and I don't think we should be committing to a stable API yet. Are we on crates.io? Hopefully we still have a chance to fix this.